### DISCUSSION TITLE
Nonlinear programming in PyOptInterface.

### DISCUSSION THREAD

**Discussion by huiq666** [metab0t/PyOptInterface#3] [General]
Dear Sir or Madam,
I would like to inquire about PyOptInterface's capabilities in handling nonlinear programming problems. I have constraint conditions similar to the following formula $\begin{aligned} & DF=\frac{\lambda_{\mathrm{H}}}{\kappa_{\mathrm{H}}\beta_{\mathrm{H}}}(\sum_{i=1,i\neq m}^{N_{\mathrm{G}}}p_{i}q_{i}L_{i}-P_{\mathrm{St,H}})+\\  & =\frac{\varepsilon_{\Sigma}\sin\theta+\sigma_{\Sigma}\cos\theta}{\kappa_{\mathrm{H}}\beta_{\mathrm{H}}\omega}\mathrm{e}^{-\alpha_{\mathrm{H}}\tau_{\mathrm{os}}}\\  & \tau_{\mathrm{os}}=\frac{\theta+\pi}{\omega}\\  & q_{i}=\mathrm{sgn}(p_{i})\end{aligned}$. I would like to know if these constraints can be written into the model using your team's PyOptInterface and solved using Gurobi.

Thank you very much!

Best regards,

**Comment by metab0t** [comment]
Your LaTeX formula is not displayed correctly.

No, PyOptInterface cannot handle nonlinear constraints in Gurobi now.

In gurobipy, the complex nonlinear constraint must be divided into simpler terms because it can only handle nonlinear constraints like $y=e^x$ and $y=sin(x)$.

If you want features like `addgenconstr` in gurobipy, we can consider implementing it.
